### PR TITLE
TST: Add a test for fillna in PeriodArray

### LINKED ISSUE #24537: Float conversion makes fillna lossy
For int64 data near the int64 implementation bounds, `astype('float64')` or `ensure_float64` is lossy.  The motivating case is `PeriodArray.fillna`

```
dti = pd.date_range(pd.Timestamp.max - pd.Timedelta(nanoseconds=10), periods=5, freq='ns')
pi = dti.to_period('ns')
parr = pi._data
parr[2] = pd.NaT

>>> parr.fillna(method='pad')
<PeriodArray>
['NaT', 'NaT', 'NaT', 'NaT', 'NaT']
Length: 5, dtype: period[N]
```

### PR DESCRIPTION
**Pull request by mstazherova**
- [x] closes #24537

### CODE REVIEW

**Comment by mstazherova** [comment]
@phofl @noatamir ready for review

File: pandas/tests/series/methods/test_fillna.py [hunk 1]
```diff
@@ -946,3 +947,27 @@ def test_datetime64tz_fillna_round_issue(self):
         )
 
         tm.assert_series_equal(filled, expected)
+
+    def test_fillna_parr(self):
+        # GH-24537
+        dti = date_range(
+            Timestamp.max - Timedelta(nanoseconds=10), periods=5, freq="ns"
+        )
+        pi = dti.to_period("ns")
```
Comment (phofl) [inline pandas/tests/series/methods/test_fillna.py:956]: You can do ser = Series(dti.to_period("ns")) here and then do

ser[2] = NaT without accessing _data in the line below, makes it a bit simpler.

And then do ser.fillna(method="pad")

Comment (mstazherova) [reply pandas/tests/series/methods/test_fillna.py:956]: Ah ok thanks will try to change it 👍

File: pandas/tests/series/methods/test_fillna.py [hunk 2]
```diff
@@ -968,6 +967,6 @@ def test_fillna_parr(self):
         )
         expected = Series(arr)
 
-        filled = Series(parr.fillna(method="pad"))
+        filled = Series(ser.fillna(method="pad"))
```
Comment (phofl) [inline pandas/tests/series/methods/test_fillna.py:970]: You can avoid Series(...) here, just do filled = ser.fillna(method="pad")

Comment (mstazherova) [reply pandas/tests/series/methods/test_fillna.py]: 🙈 🙈 this is why you shouldn't code late in the evening. Thanks, changed it ✅

Comment (mstazherova) [reply pandas/tests/series/methods/test_fillna.py]: `Install dependencies` check inside `Python Dev / actions-311-dev (macOS-latest) ` fails though...

Comment (phofl) [reply pandas/tests/series/methods/test_fillna.py]: This is unrelated, already opened an issue about it.

Comment (phofl) [reply pandas/tests/series/methods/test_fillna.py]: No not really, something will have to investigate :) But it is not related to your pr, its failing on all prs

**Comment by mstazherova** [comment]
@phofl should I still do something before this PR can be merged? Right now it's failing on pre-commit, but it's not related to my changes.

**Comment by phofl** [comment]
Sorry, was a bit busy yesterday.

pre-commit is failing, otherwise this lgtm

**Comment by phofl** [comment]
Looks like pre-commit failure came from something else, merged main to check

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @mstazherova

**Comment by mstazherova** [comment]
Cool, thank you! Excited to do more :)

**Pull request merged**